components/match2/commons/match_group_coordinates.lua

```diff
@@ -289,4 +289,69 @@ function MatchGroupCoordinates.groupMatchIdsByField(bracket, fieldName)
 	return byField
 end
 
+--[[
+Compute the number of opponents from each round onward in a bracket.
+]]
+function MatchGroupCoordinates.computeBracketOpponentCounts(bracket)
+	local countsByRound = MatchGroupCoordinates.computeRawCounts(bracket)
+
+	-- Only include positive counts
+	return Array.map(countsByRound, function(countsInRound)
+		local sum = 0
+		for _, count in ipairs(countsInRound) do
+			if count >= 0 then
+				sum = sum + count
+			end
+		end
+		return sum
+	end)
+end
+
+--[[
+Computes the number of opponents from each round onward for each section in a
+bracket. Lower bracket counts may be negative. This indicates either that an
+opponent from the upper bracket dropped down to a earlier round, or that some
+opponents leave the tournament directly from the upper bracket.
```

Comment:
This comment needs to be rewritten I think.
Also don't understand why the count can be negative even with this comment. If the count is the number of opponents still partaking in the upper/lower bracket, how can it become negative?

Reply:
the count of lower bracket rounds can be negative
due to them plus the upperbracket round together being the real value we want
basically we do not want to count the opponents from the upper bracket that drops down to the lower bracket twice